app/models/user.rb

```diff
@@ -0,0 +1,9 @@
+class User < ApplicationRecord
+  def self.all_user_public_data
+    all.select(:id, :name, :created_at, :updated_at)
+  end
+
+  def self.single_user_public_data(id)
+    select(:id, :name, :created_at, :updated_at).find(id)
+  end
```

Comment:
I'm ok with leaving this here for now since I guess the api will change anyways. But a couple of things to think about:
 - how can we get rid of the duplication?
 - is it an User's responsibility to know what's considered "public_data" and what we want to expose through the API? What if we have different versions of "public data" (like a public profile view or an admin's view).

Reply:
👍 to @ijpiantanida 

This is definitely an improvement, and it addresses almost all of the concerns I originally listed, however one still has to look at the `User` class in order to see what is being exposed from the API.

I think looking at it in terms of responsibilities is a good idea. To ask a somewhat leading question, who should be responsible for determining what the API exposes?

Reply:
I too am OK with leaving it here for now.

Reply:
I implemented the serializer and was planning on removing the `select` from the models, but I'm not sure how I'd get data that isn't explicitly spelled out in the serializer (ie if I added an `admin` route).

@ijpiantanida I was thinking that the 'user' here would be someone consuming data from the API, so they'd only be exposed to the routes, and not see what's going on in the controller / models, but maybe I'm thinking about it wrong.

@ericmeyer I was thinking that if I removed the `select` from the User model, then those values would only exist in the serializer, but that still leaves your concern of a user having to look somewhere to find out why the route isn't returning everything they might expect. In your scenario, would a 'user' of my app be another developer looking to add features?